presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -197,7 +197,7 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
         }
         Map<ColumnHandle, NullableValue> values = builder.build();
 
-        if (!constraint.predicate().test(values)) {
+        if (constraint.predicate().isPresent() && !constraint.predicate().get().test(values)) {
```

Comment:
> Optional.empty() is much better because connector can avoid such
> filtering which in some cases could be very expensive.

What are the cases?

Reply:
- https://github.com/prestodb/presto/pull/10336 where for `bindings -> true` all schemas and all tables are populated and tested
 - also in hive connector in case of empty predicate there might be possible to avoid (postpone) partitions retrieval from metastore

Reply:
> #10336 where for bindings -> true all schemas and all tables are populated and tested

Notice that this usually happens 4 times in `PickTableLayout` per query

Reply:
This is to about changing SPI. I do not want to go too far just in case this change got rejected.

Anyway, even SPI change is my opinion worth doing even if nothing else changed so far.

Reply:
I understand. OTOH, it's better not to change SPI at all, if we do not reap the benefits.